text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+a host (such as `rustc` and `cargo`). Such a platform must meet all the
+requirements above, and must additionally get the compiler and infrastructure
+team to approve the building of host tools.
+
+## Tier 1 target policy
+
+At this tier, the Rust project guarantees that a target builds and passes all
+tests, and will reject patches that fail to build or pass the testsuite on a
+target. We hold tier 1 targets to our highest standard of requirements.
+
+- Any new tier 1 target requires compiler team approval based on these
+  requirements.
+- In addition, the infrastructure team must approve the integration of the
+  target into CI, and the CI-related requirements. This review and approval
+  will typically take place in the PR adding the target to CI.
+- In addition, the release team must approve the long-term viability of the
+  target, and the additional work of supporting the target.
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
+  users of Rust across multiple organizations or projects. A tier 1 target may
+  be demoted or removed if it becomes obsolete or no longer meets this
+  requirement.
+- The target must build and pass tests reliably in CI.
+- Building the target and running the testsuite for the target must not take
+  substantially longer than other targets.
+- If running the testsuite requires additional infrastructure (such as systems
+  running the target), the target development team shall arrange to provide
+  such resources to the Rust project, to the satisfaction and approval of the
+  Rust infrastructure team.
+- Tier 1 targets must provide documentation for the Rust community for how to
+  build and run tests for the platform, using emulation if possible, or
+  dedicated hardware if necessary.
+- A tier 1 target may be demoted or removed if it no longer meets these
+  requirements. Any proposal for demotion or removal will be communicated
+  widely to the Rust community, both when initially proposed and before being
+  dropped from a stable release.
+- All tier 2 requirements apply.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+While these criteria attempt to document the policy, that policy still involves
+human judgment. Targets must fulfill the spirit of the requirements as well, as
+determined by the judgment of the approving teams.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The set of approving teams for each tier arose out of discussion with the
+various teams involved with aspects of the Rust project impacted by new
+targets.
+
+
+# Prior art
+[prior-art]: #prior-art
+
+This attempts to formalize and document Rust policy around targets and
+architectures. https://forge.rust-lang.org/release/platform-support.html
+documents some of that policy.
+
+Future expansions of such policy may find requirements from other communities
```

Comment:
I assume Rust's tiered approach is derived from [firefox's tiers](https://developer.mozilla.org/en-US/docs/Mozilla/Supported_build_configurations), is that true?  I tried to find other languages/tools using a similar tiered approach, but only found [node](https://github.com/nodejs/node/blob/master/BUILDING.md#supported-platforms) had something similar.  I know sometimes in web development projects will have "grades" for browser support.

Reply:
https://gitlab.haskell.org/ghc/ghc/wikis/platforms

Reply:
Nice finds! I've added links.